src/esc/environment_fs_provider.ts

```diff
@@ -60,12 +62,20 @@ export class EnvironmentFileSystemProvider implements vscode.FileSystemProvider,
             };
         }
 
-        if (parts.includes('open')) {
-            const uriStr = uri.toString();
-            const index = uriStr.lastIndexOf('/open');
-            const envPath = uriStr.slice(0, index);
-            const mtime = this.updates.get(envPath) || 0;
-            const size = this.sizes.get(uri.toString()) || 0;
+        let uriStr = uri.toString();
+        const isOpen = parts.includes('open');
+        if (isOpen) {
+            uriStr = this.getOpenEnvPrefix(uri.toString());
+        }
+        
+        if (!this.updates.has(uriStr)) {
+            this.updates.set(uriStr, Date.now());
+        }
+
+        const mtime = this.updates.get(uriStr) || 0;
+        const size = this.sizes.get(uriStr) || randomInt(1000);
```

Comment:
The fake values would cause it to reload the yaml far more often.  We want the fake values to be there when we know there is a change and "consistent" values so it does not re-read the env over and over.